### ISSUE TITLE
NVE simulations don't seem to benefit from `jit`

### ISSUE THREAD

**Issue by fabiannagel** [google/jax-md#169]
`jit` works very well for accelerating energy and force functions but I found that it does not seem to do much for simulations.
I first noticed this when running my own NVE benchmarks, but I was able to reproduce it based on the LJ benchmark notebook from this repo. You can find it [here](https://gist.github.com/fabiannagel/be98fa04b4ba3d51395c782425031824).

TLDR:
- `jit` with Lennard-Jones accelerates energy and force calculations by approx. factors of 7 to 10.
- However, I can't get any acceleration for NVE simulations.
- Whether I jit the `step_fn`, `apply_fn` or `energy_fn` seems to make no difference.

Maybe this is something that you have already observed yourself? Also, I'd be curious to hear why you chose to not use `jit` in the NVE examples. So far, my understanding of `jit` has been that it is not equivalent to performing warm-up runs, right?

Looking forward to hearing your thoughts on this!

Best
Fabian

**Comment by fabiannagel** [comment]
Turns out certain `jax.lax` primitives `jit` by default, and `lax.fori_loop()` would be an example for that. Unfortunately, this doesn't seem to be mentioned anywhere in the JAX docs.

With that in mind, it makes sense that (implicitly jitted) NVE step times are very close to the jitted runtime of a single force evaluation.
